tools/gcs-usage-recommender/README.md

```diff
@@ -0,0 +1,247 @@
+export AUDIT_LOG_PROJECT_ID=gcsauditlogs
+export AUDIT_LOG_DATASET_ID=gcs_audit_logs
+export OUTPUT_PROJECT_ID=gcsusagerecommder
+export OUTPUT_DATASET_ID=gcs_usage_log
+export OUTPUT_TABLE_NAME=gcs_bucket_insights
+export BQ_LOCATION=EU
+export ORG_ID=622243302570
+````
+* `AUDIT_LOG_PROJECT_ID` is the project ID in BigQuery where you will export the
+audit logs.
+* `AUDIT_LOG_DATASET_ID` is the dataset ID in BigQuery where the audit logs will
+be exported to.
+* `OUTPUT_PROJECT_ID` is the project ID where the bucket insights table will be.
+* `OUTPUT_DATASET_ID` is the dataset ID where the bucket insights table will be.
+* `OUTPUT_TABLE_NAME` is the name of the BigQuery table holding the bucket insights.
+* `BQ_LOCATION` is either EU or US, and will be for both datasets created.
+* `ORG_ID` Find yours by invoking `gcloud organizations list`
+
+<h3> 2.2 Permissions </h3>
+To follow these interactive instructions, you will need permissions to:
+
+* Create and manage a BigQuery dataset in the project you’ve chosen
+  * bigquery.datasets.create
+  * bigquery.tables.create
+  * bigquery.tables.getData
+  * bigquery.tables.updateData
+
+And you will need permissions at the organization level to:
+
+* Create an organization-level audit log sink
+  * roles/logging.configWriter
+
+* Scan your inventory:
+  * storage.buckets.get
+  * storage.buckets.list
+  * storage.objects.get
+  * storage.objects.list
+  * resourcemanager.organizations.get
+  * resourcemanager.projects.get
+  * resourcemanager.projects.list
+
+You will need these permissions only during initial setup.
+
+<h3> 2.3 Preparing the audit log dataset</h3>
+Let’s walk through the steps necessary to create and populate the dataset.
+
+<h4> 2.3.1 Creating the dataset </h4>
+Let’s create the dataset inside the project you specified above. 
+<b>Note that this does not specify a partition expiration. Audit logs can get
+costly, so it is recommended to choose a default partition expiration for this
+dataset, such as 90 days.</b>
+
+````bq mk --location=${BQ_LOCATION} -d "${OUTPUT_PROJECT_ID}:${OUTPUT_DATASET_ID}"````
+
+<h4> 2.3.2 Create the audit log sink </h4>
+
+For GCS, we'll be enabling two types of [audit logs](https://cloud.google.com/storage/docs/audit-logs):
+* <b>Admin Activity logs</b>: Entries for operations that modify the 
+configuration or metadata of a project, bucket, or object.
+ 
+* <b>Data Access logs</b>: Entries for operations that modify objects or read a 
+project, bucket, or object. There are several sub-types of data access logs:
+    * <b>ADMIN_READ</b>: Entries for operations that read the configuration or 
+    metadata of a project, bucket, or object.
+    * <b>DATA_READ</b>: Entries for operations that read an object.
+    * <b>DATA_WRITE</b>: Entries for operations that create or modify an object.
+
+By default, only admin activity logs are enabled for organizations. However, data 
+access logs are not configured by default. This will require us to enable them in 

```

Comment:
"require" is a strong word.
can this really not be achieved w/ API Call / gsutil / terraform?
Is this because you're turning it on at org level?